main.go

```diff
@@ -1,11 +1,53 @@
 package main
 
 import (
+	"log"
 	_ "net/http/pprof" //nolint:gosec
 
+	"github.com/kardianos/service"
 	"github.com/navidrome/navidrome/cmd"
 )
 
-func main() {
+var logger service.Logger
+
+type program struct{}
+
+func (p *program) Start(s service.Service) error {
+	go p.run()
+	return nil
+}
+func (p *program) run() {
 	cmd.Execute()
 }
+func (p *program) Stop(s service.Service) error {
+	return nil
+}
+
+func main() {
+	if service.Interactive() {
```

Comment:
I don't quite understand why you had to do this. In my original branch, I was able to add all the service logic in the `cmd/svc.go` file: https://github.com/navidrome/navidrome/compare/master...service

Ofc, I didn't test in Windows, so not sure if it has to do with it.

Reply:
So the original implementation basically only had the service _management_ within it, it would ask systemd/windows or whatever to start the service (which is basically just the executable running in a different environment). A service in windows is not "just a process that runs", there are specific API callbacks that the application needs to run when it's executed as a service to indicate to the service manager that it has started correctly. This basically means that you need to wrap the whole actual executable so that Kardianos can handle those callbacks. If you chose to _always_ do the non-interactive route (i.e. wrap it all the time) then the application never returns as it'll be waiting for SIGHUP or whichever callback needs to executed for Windows (Kardianos handles that for you).

We can absolutely refactor this, but fundamentally we will need to have a flow similar to this, in `main`.

Reply:
As discussed on Github, I've basically bombed this out and moved it into `navidrome service execute` and changed the msi to do that.